### ISSUE TITLE
Search button overlaps content in smaller viewports

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#11121]
## Issue Description
The search form, "Can't find…", overlap in smaller viewports 

## Steps to Recreate
Enter https://www.va.gov/find-locations in browser
Adjust screensize below 1000 pixels width and observe the search button. The button moves out of the search box to overlap the text above the map. 

![image](https://user-images.githubusercontent.com/55411834/87177704-4f090f00-c299-11ea-9d17-005c8e3fbace.png)
![image](https://user-images.githubusercontent.com/55411834/87177750-62b47580-c299-11ea-973b-338dc55187df.png)

## Front end tasks
- [ ] Correct responsiveness to ensure all content is visible and does not overlap in any screen width

To ensure existing functionality is maintained: 
- [x] Run axe checks using the Chrome or Firefox browser plugin
- [x] Test for color contrast and color blindness issues
- [x] Zoom layouts to 400% at 1280px width
- [ ] Test with 1 or 2 screen readers
- [x] Navigate using the keyboard only

## Acceptance Criteria
- [ ] Designer comments on this issue to approve page styling 
- [x] All content is visible and does not overlap in any screen width
- [x] All axe checks pass
- [x] All color contrast checks pass
- [x] All zoom testing passes
- [x] All keyboard checks pass 
- [ ] All screenreader checks pass

## Appendix
<details> 

 <summary>Ensure your code changes are covered by E2E tests (expand)</summary>
 
 - Add E2E tests if none exist for this addition/change. 
 
 - Update existing E2E tests if this code will change functionality. 
 
 - Include axe checks, including hidden content
 
</details>

<details> 

 <summary> Run axe checks using the Chrome or Firefox browser plugin (expand)</summary>
 
- Ensure no heading levels are skipped.

- Ensure all buttons and labeled inputs use semantic HTML elements.

- Ensure all buttons, labeled elements and images are identified using HTML semantic markers or ARIA roles.

- Ensure form fields have clearly defined boundaries or outlines.

- Ensure form fields do not use placeholder text. 

- Ensure form fields have highly visible and specific error states. 

</details>

<details> 

 <summary> Test for color contrast and color blindness issues (expand) </summary>
 - All text has appropriate contrast. 

</details>

<details> 

 <summary> Zoom layouts to 400% at 1280px width (expand)</summary> 
 - Ensure readability and usability are supported when zoomed up to 400% at 1280px browser width
 - Ensure no content gets focused offscreen or is hidden from view.
 
</details>

<details> 

 <summary> Test with 1 or 2 screen readers (expand)</summary> 
 
 - Ensure the page includes a skip navigation link. 
 
 - Ensure all links are properly descriptive. 
 
 - Ensure screen reader users can hear the text equivalent for each image conveying information. 

 - Ensure screen reader users can hear the text equivalent for each image button. 

 - Ensure screen reader users can hear labels and instructions for inputs. 
 
 - Ensure purely decorative images are not announced by the screenreader. 
 
</details>

<details> 

 <summary>Navigate using the keyboard only (expand)</summary> 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be reached with the tab key in a logical order. 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure all interactive elements can be reached with the tab key in a logical order 
 
 - Ensure all interactive elements can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure focus is always visible and appears in logical order.
 
 - Ensure each interactive element has visible focus state which appears in logical order.
 

</details>

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by andaleliz** [comment]
Just FYI - on an iPhone 10 landscape view, the button doesn't overlap content, but it sits way off to the right side and is slightly coming out of the gray container.  Button doesn't appear to be problematic on iPhone 10 portrait view.

**Comment by mmiddaugh** [comment]
"Can't find..." has been moved to below the map. 

@jenstrickland can you please validate?

**Comment by jenstrickland** [comment]
It no longer overlaps the "Can't find text, but now the button breaks out of the form area and the map overlaps it. Should we close this 508-defect-1 issue and create a 508-defect-2 issue for the map overlap? 

![Screen Shot 2020-07-30 at 7 44 15 PM](https://user-images.githubusercontent.com/57469/88984946-15766300-d29d-11ea-8cc2-219fe5e48c12.png)

**Comment by jenstrickland** [comment]
Actually, I think we should just keep this issue open and continue to work to address it, as the Search button is cut off the right side of the screen at some widths.

![Screen Shot 2020-07-30 at 7 52 40 PM](https://user-images.githubusercontent.com/57469/88985436-3ee3be80-d29e-11ea-9e1a-1296bb60f4ea.png)

**Comment by mmiddaugh** [comment]
I support your suggestion, @jenstrickland.  Let's leave it open.

**Comment by mpelzsherman** [comment]
Should we move this back to Backlog?

**Comment by jenstrickland** [comment]
@mpelzsherman Since this is a 508-defect-1, it should be addressed ASAP.

**Comment by mpelzsherman** [comment]
@lildziner is working on designs for the Facility Locator to make it work at all breakpoints. Once that work is done, we can proceed to implement and address this issue.

**Comment by jenstrickland** [comment]
@mpelzsherman Due to this being a [508-defect-1 issue](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1), we ought to address this sooner than later, and I leave it to @mmiddaugh to know where it makes sense to address this.

**Comment by lildziner** [comment]
@mpelzsherman I think the button issue is addresses in [all the breakpoints in the UXPin prototype](https://preview.uxpin.com/f373333a8a59902cbf120c3f00e6d0e13d52fb75#/pages/131155292/simulate/sitemap?mode=i). Let me know if you have any questions about any specifics.

**Comment by lildziner** [comment]
![image](https://user-images.githubusercontent.com/4960080/89104870-7c556280-d3ea-11ea-904d-89c959d696b8.png) Here's a view. Ideally the inputs, select menus and button stack and go fullwidth at all the smaller widths.

**Comment by jenstrickland** [comment]
On Monday, we can see if @mmiddaugh wants to move forward with this. @lildziner - have you defined the viewport width this should happen at? 

This issue happens at just under 1000 pixels wide, which seems too early to stack like the above design. We also have min- and max-widths established for inputs and buttons. What might a layout look like between 1000 and 600 pixels wide, 600 and 400 pixels wide, etc? :)

**Comment by mmiddaugh** [comment]
Moving this back to Icebox. Our work in [#12131] should resolve this issue. We can ask Jennifer to validate it once that issue is complete.

**Comment by mmiddaugh** [comment]
@1Copenut if you are available to validate this one, I think our responsive design changes will have resolved the issue.

**Comment by 1Copenut** [comment]
@mmiddaugh I tested this and am pasting a screen shot at 925px width from Chrome. It follows the designer's recommended behavior. Moving to close.

---
<img width="880" alt="Screen Shot 2020-09-02 at 11 56 49 AM" src="https://user-images.githubusercontent.com/934879/92013787-d9a74100-ed13-11ea-8dae-b8b1c809c2e1.png">
